From 8fb8ec4eca06e7f58151c91e0d2a07bbc687152a Mon Sep 17 00:00:00 2001
From: MrFixIt2001 <mrfixit2001@gmail.com>
Date: Tue, 23 Jun 2020 15:38:27 -0400
Subject: [PATCH] rk3399 dtsi vop clock Rockchip's vop-clk-set dtsi for rk3399
 had some clocks duplicated with two different speeds. Cleanup and alphabetize
 this list so it's managable. Saved both previous clock rates in the comments
 for fututure reference.

---
 .../boot/dts/rockchip/rk3399-vop-clk-set.dtsi | 181 ++++++++++--------
 1 file changed, 106 insertions(+), 75 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-vop-clk-set.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-vop-clk-set.dtsi
index 5ed8dac6cf..83c32b1d45 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-vop-clk-set.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-vop-clk-set.dtsi
@@ -99,83 +99,114 @@
 		assigned-clock-parents = <&cru PLL_GPLL>;
 };
 
+/* Rockchip had a bunch of duplicate clocks with different rates defined.
+ * I have removed the duplicates, using only the one that was assigned last, and put them in alphabetical order.
+ * Below are both values for the duplicated clocks for future reference:
+ *   ACLK_CCI = 150000000, 600000000
+ *   ACLK_CENTER = 100000000, 400000000
+ *   ACLK_EMMC = 50000000, 100000000
+ *   ACLK_GMAC = 100000000, 400000000
+ *   ACLK_PERIHP = 150000000, 75000000
+ *   ACLK_PERILP0 = 50000000, 300000000
+ *   ACLK_USB3 = 150000000, 300000000
+ *   FCLK_CM0S = 50000000, 300000000
+ *   HCLK_PERILP1 = 100000000, 50000000
+ *   HCLK_SD = 50000000, 200000000
+ *   PCLK_ALIVE = 50000000, 100000000
+ *   SCLK_CCI_TRACE = 75000000, 150000000
+ *   SCLK_CS = 150000000, 75000000
+ *   SCLK_RGA_CORE = 400000000
+ *   SCLK_VDU_CA = 150000000, 300000000
+ */
 &cru {
 		assigned-clocks =
-			<&cru ACLK_PERIHP>, <&cru ACLK_PERILP0>,
-			<&cru HCLK_PERILP1>, <&cru SCLK_SDMMC>,
-			<&cru ACLK_EMMC>, <&cru ACLK_CENTER>,
-			<&cru HCLK_SD>, <&cru SCLK_VDU_CA>,
-			<&cru SCLK_VDU_CORE>, <&cru ACLK_USB3>,
-			<&cru FCLK_CM0S>, <&cru ACLK_CCI>,
-			<&cru PCLK_ALIVE>, <&cru ACLK_GMAC>,
-			<&cru SCLK_CS>, <&cru SCLK_CCI_TRACE>,
-			<&cru ARMCLKL>, <&cru ARMCLKB>,
-			<&cru PLL_NPLL>, <&cru ACLK_GPU>,
-			<&cru PLL_GPLL>, <&cru ACLK_PERIHP>,
-			<&cru HCLK_PERIHP>, <&cru PCLK_PERIHP>,
-			<&cru ACLK_PERILP0>, <&cru HCLK_PERILP0>,
-			<&cru PCLK_PERILP0>, <&cru HCLK_PERILP1>,
-			<&cru PCLK_PERILP1>, <&cru SCLK_I2C1>,
-			<&cru SCLK_I2C2>, <&cru SCLK_I2C3>,
-			<&cru SCLK_I2C5>, <&cru SCLK_I2C6>,
-			<&cru SCLK_I2C7>, <&cru SCLK_SPI0>,
-			<&cru SCLK_SPI1>, <&cru SCLK_SPI2>,
-			<&cru SCLK_SPI4>, <&cru SCLK_SPI5>,
-			<&cru ACLK_GIC>, <&cru ACLK_ISP0>,
-			<&cru ACLK_ISP1>, <&cru SCLK_VOP0_PWM>,
-			<&cru SCLK_VOP1_PWM>, <&cru PCLK_EDP>,
-			<&cru ACLK_HDCP>, <&cru ACLK_VIO>,
-			<&cru HCLK_SD>, <&cru SCLK_CRYPTO0>,
-			<&cru SCLK_CRYPTO1>, <&cru SCLK_EMMC>,
-			<&cru ACLK_EMMC>, <&cru ACLK_CENTER>,
-			<&cru ACLK_IEP>, <&cru ACLK_RGA>,
-			<&cru SCLK_RGA_CORE>, <&cru ACLK_VDU>,
-			<&cru ACLK_VCODEC>, <&cru PCLK_DDR>,
-			<&cru ACLK_GMAC>, <&cru SCLK_VDU_CA>,
-			<&cru SCLK_VDU_CORE>, <&cru ACLK_USB3>,
-			<&cru FCLK_CM0S>, <&cru ACLK_CCI>,
-			<&cru PCLK_ALIVE>, <&cru SCLK_CS>,
-			<&cru SCLK_CCI_TRACE>, <&cru ACLK_VOP0>,
-			<&cru HCLK_VOP0>, <&cru ACLK_VOP1>,
-			<&cru HCLK_VOP1>;
+			 <&cru ACLK_CCI>, <&cru ACLK_CENTER>,
+			 <&cru ACLK_EMMC>, <&cru ACLK_GIC>,
+			 <&cru ACLK_GMAC>, <&cru ACLK_GPU>,
+			 <&cru ACLK_HDCP>, <&cru ACLK_IEP>,
+			 <&cru ACLK_ISP0>, <&cru ACLK_ISP1>,
+			 <&cru ACLK_PERIHP>, <&cru ACLK_PERILP0>,
+			 <&cru ACLK_RGA>, <&cru ACLK_USB3>,
+			 <&cru ACLK_VCODEC>, <&cru ACLK_VDU>,
+			 <&cru ACLK_VIO>, <&cru ACLK_VOP0>,
+			 <&cru ACLK_VOP1>, <&cru ARMCLKB>,
+			 <&cru ARMCLKL>, <&cru FCLK_CM0S>,
+			 <&cru HCLK_PERIHP>, <&cru HCLK_PERILP0>,
+			 <&cru HCLK_PERILP1>, <&cru HCLK_SD>,
+			 <&cru HCLK_VOP0>, <&cru HCLK_VOP1>,
+			 <&cru PCLK_ALIVE>, <&cru PCLK_DDR>,
+			 <&cru PCLK_EDP>, <&cru PCLK_PERIHP>,
+			 <&cru PCLK_PERILP0>, <&cru PCLK_PERILP1>,
+			 <&cru PLL_GPLL>, <&cru PLL_NPLL>,
+			 <&cru SCLK_CCI_TRACE>, <&cru SCLK_CRYPTO0>,
+			 <&cru SCLK_CRYPTO1>, <&cru SCLK_CS>,
+			 <&cru SCLK_EMMC>, <&cru SCLK_I2C1>,
+			 <&cru SCLK_I2C2>, <&cru SCLK_I2C3>,
+			 <&cru SCLK_I2C5>, <&cru SCLK_I2C6>,
+			 <&cru SCLK_I2C7>, <&cru SCLK_RGA_CORE>,
+			 <&cru SCLK_SDMMC>, <&cru SCLK_SPI0>,
+			 <&cru SCLK_SPI1>, <&cru SCLK_SPI2>,
+			 <&cru SCLK_SPI4>, <&cru SCLK_SPI5>,
+			 <&cru SCLK_VDU_CA>, <&cru SCLK_VDU_CORE>,
+			 <&cru SCLK_VOP0_PWM>, <&cru SCLK_VOP1_PWM>;
 		assigned-clock-rates =
-			 <75000000>, <50000000>,
-			 <50000000>, <50000000>,
-			 <50000000>, <100000000>,
-			 <50000000>, <150000000>,
-			 <150000000>, <150000000>,
-			 <50000000>, <150000000>,
-			 <50000000>, <100000000>,
-			 <75000000>, <75000000>,
-			 <816000000>, <816000000>,
-			 <600000000>, <200000000>,
-			 <800000000>, <150000000>,
-			 <75000000>, <37500000>,
-			 <300000000>, <100000000>,
-			 <50000000>, <100000000>,
-			 <50000000>, <100000000>,
-			 <100000000>, <100000000>,
-			 <100000000>, <100000000>,
-			 <100000000>, <50000000>,
-			 <50000000>, <50000000>,
-			 <50000000>, <50000000>,
-			 <200000000>, <400000000>,
-			 <400000000>, <100000000>,
-			 <100000000>, <100000000>,
-			 <400000000>, <400000000>,
-			 <200000000>, <100000000>,
-			 <200000000>, <200000000>,
-			 <100000000>, <400000000>,
-			 <400000000>, <400000000>,
-			 <400000000>, <300000000>,
-			 <400000000>, <200000000>,
-			 <400000000>, <300000000>,
-			 <300000000>, <300000000>,
-			 <300000000>, <600000000>,/* aclk_cci */
-			 <100000000>, <150000000>,
-			 <150000000>, <400000000>,
-			 <100000000>, <400000000>,
-			 <100000000>;
+			 /*ACLK_CCI*/ <600000000>,
+			 /*ACLK_CENTER*/ <400000000>,
+			 /*ACLK_EMMC*/ <100000000>,
+			 /*ACLK_GIC*/ <200000000>,
+			 /*ACLK_GMAC*/ <400000000>,
+			 /*ACLK_GPU*/ <200000000>,
+			 /*ACLK_HDCP*/ <400000000>,
+			 /*ACLK_IEP*/ <400000000>,
+			 /*ACLK_ISP0*/ <400000000>,
+			 /*ACLK_ISP1*/ <400000000>,
+			 /*ACLK_PERIHP*/ <150000000>,
+			 /*ACLK_PERILP0*/ <300000000>,
+			 /*ACLK_RGA*/ <400000000>,
+			 /*ACLK_USB3*/ <300000000>,
+			 /*ACLK_VCODEC*/ <400000000>,
+			 /*ACLK_VDU*/ <300000000>,
+			 /*ACLK_VIO*/ <400000000>,
+			 /*ACLK_VOP0*/ <400000000>,
+			 /*ACLK_VOP1*/ <400000000>,
+			 /*ARMCLKB*/ <816000000>,
+			 /*ARMCLKL*/ <816000000>,
+			 /*FCLK_CM0S*/ <300000000>,
+			 /*HCLK_PERIHP*/ <75000000>,
+			 /*HCLK_PERILP0*/ <100000000>,
+			 /*HCLK_PERILP1*/ <100000000>,
+			 /*HCLK_SD*/ <200000000>,
+			 /*HCLK_VOP0*/ <100000000>,
+			 /*HCLK_VOP1*/ <100000000>,
+			 /*PCLK_ALIVE*/ <100000000>,
+			 /*PCLK_DDR*/ <200000000>,
+			 /*PCLK_EDP*/ <100000000>,
+			 /*PCLK_PERIHP*/ <37500000>,
+			 /*PCLK_PERILP0*/ <50000000>,
+			 /*PCLK_PERILP1*/ <50000000>,
+			 /*PLL_GPLL*/ <800000000>,
+			 /*PLL_NPLL*/ <600000000>,
+			 /*SCLK_CCI_TRACE*/ <150000000>,
+			 /*SCLK_CRYPTO0*/ <100000000>,
+			 /*SCLK_CRYPTO1*/ <200000000>,
+			 /*SCLK_CS*/ <150000000>,
+			 /*SCLK_EMMC*/ <200000000>,
+			 /*SCLK_I2C1*/ <100000000>,
+			 /*SCLK_I2C2*/ <100000000>,
+			 /*SCLK_I2C3*/ <100000000>,
+			 /*SCLK_I2C5*/ <100000000>,
+			 /*SCLK_I2C6*/ <100000000>,
+			 /*SCLK_I2C7*/ <100000000>,
+			 /*SCLK_RGA_CORE*/ <400000000>,
+			 /*SCLK_SDMMC*/ <50000000>,
+			 /*SCLK_SPI0*/ <50000000>,
+			 /*SCLK_SPI1*/ <50000000>,
+			 /*SCLK_SPI2*/ <50000000>,
+			 /*SCLK_SPI4*/ <50000000>,
+			 /*SCLK_SPI5*/ <50000000>,
+			 /*SCLK_VDU_CA*/ <300000000>,
+			 /*SCLK_VDU_CORE*/ <300000000>,
+			 /*SCLK_VOP0_PWM*/ <100000000>,
+			 /*SCLK_VOP1_PWM*/ <100000000>;
 };
 #endif
-

